p4c_bm/gen_json.py

```diff
@@ -1198,6 +1210,30 @@ def dump_field_aliases(json_dict, hlir, path_field_aliases):
     field_aliases = [[a, v] for a, v in aliases_dict.items()]
     json_dict["field_aliases"] = field_aliases
 
+def dump_extern_instances(json_dict, hlir):
+    extern_instances = []
+    id_ = 0
+    for name, p4_extern_instance in hlir.p4_extern_instances.items():
+        extern_instance_dict = OrderedDict()
+        extern_instance_dict["name"] = name
+        extern_instance_dict["id"] = id_
+        extern_instance_dict["type"] = p4_extern_instance.extern_type.name
+
+        id_ += 1
+
+        attributes = []
+        for attribute, attr in p4_extern_instance.attributes.items():
+            attribute_dict = OrderedDict()
+            attribute_dict["name"] = attribute
+            attribute_dict["type"] = "hexstr" # Ref: P4Objects.cpp, line 264
+            attribute_dict["value"] = attr.strip()
```

Comment:
As you correctly noted, only `hexstr` attributes are supported in bmv2 (currently). As a result you should probably check that `attr` has the right type and issue a `LOG_CRITICAL` if it is not. Btw, you seem to assume that `attr` is a string; but I would believe an integer value would make more sense. `hexstr` is supposed to represent an integer value. What kind of attribute type did you use in your P4 programs; did you use `string`? I think that given the current extent of `extern` support in bmv2, it would make more sense to use integer attributes (`bit` or `int` type?).